caikit/core/data_model/dataobject.py

```diff
@@ -270,31 +292,32 @@ def _get_all_enums(
     return all_enums
 
 
-def _make_data_model_class(proto_class, wrapped_cls):
-    wrapper_cls = _DataBaseMetaClass(
-        wrapped_cls.__name__,
-        tuple([DataBase, wrapped_cls]),
-        {"_proto_class": proto_class, **wrapped_cls.__dict__},
-    )
-    update_wrapper(wrapper_cls, wrapped_cls, updated=())
+def _make_data_model_class(proto_class, cls):
+    if issubclass(cls, DataObjectBase):
+        _DataBaseMetaClass.parse_proto_descriptor(cls)
 
     # Recursively make all nested message wrappers
     for nested_message_descriptor in proto_class.DESCRIPTOR.nested_types:
         nested_message_name = nested_message_descriptor.name
         nested_proto_class = getattr(proto_class, nested_message_name)
         setattr(
-            wrapper_cls,
+            cls,
             nested_message_name,
             _make_data_model_class(
                 nested_proto_class,
-                types.new_class(nested_message_name),
+                _DataBaseMetaClass.__new__(
```

Comment:
There might be a cleaner way to do this. The doc on `types.new_class` says that its whole reason to exist is to create a new type using the right `metaclass`, but I couldn't easily find a way to give it `attrs`, so I wend the manual route.